src/engine/shared/snapshot.cpp

```diff
@@ -743,7 +745,10 @@ int CSnapshotBuilder::GetTypeFromIndex(int Index) const
 
 void CSnapshotBuilder::AddExtendedItemType(int Index)
 {
-	dbg_assert(0 <= Index && Index < m_NumExtendedItemTypes, "index out of range");
+	dbg_assert(0 <= Index && Index <= m_NumExtendedItemTypes, "index out of range");
```

Comment:
```suggestion
	dbg_assert(0 <= Index && Index < m_NumExtendedItemTypes, "index out of range");
```

Please increment `m_NumExtendedItemTypes` first so that we can keep this out-of-bounds assert.